src/dwn.ts

```diff
@@ -110,6 +116,7 @@ export class Dwn {
 
 export type DwnConfig = {
   didResolver?: DidResolver,
+  createStore?: typeof createStore,
```

Comment:
the downside of not having it here means that if someone wants the default behavior but just with a different `Level` under the hood then they have to go through the trouble of importing and creating their own `MessageStoreLevel` and `DataStoreLevel`
```js
import { DataStoreLevel, Dwn, MessageStoreLevel } from 'dwn-sdk-js';

Dwn.create({
    messageStore: new MessageStoreLevel({ createStore }),
    dataStore: new DataStoreLevel({ createStore }),
})
```

whereas with this approach it's much simpler/cleaner
```js
import { Dwn } from 'dwn-sdk-js';

Dwn.create({ createStore })
```

Reply:
I agree that the number of lines needed is clearly less if `createDatabase` is known by `Dwn`, but have to disagree that it is therefore better.

While often less code = better code, we know it is not always the case.

There are a couple of layers of indirect knowledge that a reader/user of the line `Dwn.create({ createDatabase })` have to know "somehow" (perhaps through documentation, which is currently lacking):
* the config param `createDatabase` is intended for specifically for the built-in **level** implementations;
* the config param is meaningless to other implementations
* the param affects both message and datastore at the same time
* you'd have to do it the longer way if you want to customize one store but not the other

Where as the similar intent of `createDatabase` in the line `dataStore: new DataStoreLevel({ createDatabase })` is much easier to comprehend: oh, `createDatabase` is used by the level store constructor, if I care enough, I will drill in. I don't have to "wonder" if I'd be missing important context by glossing over `createDatabase`.

Beyond the above, I still want to go back to the original point on separation of concern, `Dwn` shouldn't care/know about what any particular implementation of any interface need for instantiation, that's why it defined those interfaces to begin with. An analogy would be that we also throw in a config param for the ION resolver in `DwnConfig`; since `didResolver` like the stores is a param in `DwnConfig`, we throw in one more for the cache used by the DID resolver as well, maybe a couple more a few months down the road for some other components... you get the idea, soon we'd be asking ourselves how did `DwnConfig` become so "messy"...

All is say that I still hope to convince you to not expose the param at `Dwn` level for the time being.

Reply:
while i understand your point about separation of concerns, i'd argue it's making for a worse developer experience

not a hill imma die on tho, so ill remove this bit